dir.c

```diff
@@ -1461,10 +1461,10 @@ enum pattern_match_result path_matches_pattern_list(
 	 */
 	if (parent_pathname.len > 0 &&
 	    parent_pathname.buf[parent_pathname.len - 1] == '/') {
-		slash_pos = parent_pathname.buf + parent_pathname.len - 1;
+		slash_pos = parent_pathname.len - 1;
 		strbuf_add(&parent_pathname, "-", 1);
 	} else {
-		slash_pos = strrchr(parent_pathname.buf, '/');
+		slash_pos = strrchr(parent_pathname.buf, '/') - parent_pathname.buf;
```

Comment:
strrchr() can return NULL if the string does not contain a slash.

Reply:
thanks! the point of changing this was to be more robust, but still I messed it up. All better now.